src/RBF/rbf.hpp

```diff
@@ -64,6 +66,28 @@ enum class RBFMode {
 };
 
 class RBFKernel{
+    /*!
+     * Linear Polynomial class. A linear multivariate polynomial in function of the 
+     * variables defining the RBF nodes is used to regularize the interpolation 
+     * problem through RBFs. Used only when RBFMode::PARAM is enabled. 
+     * The LinearPolynomial class derives from ReconstructionPolynomial class
+     * that is intrisecally defined to work in a three dimensional space.
+     */
+    class LinearPolynomial : public ReconstructionPolynomial
+    {
+private:
+        int m_dim;                          /**< Polynomial variables space dimension */
+        int m_fields;                       /**< Number of equations of the polynomial object */
+
+    public:
+        LinearPolynomial();
+        void setDefault();
+        void clear();
```

Comment:
What's the point of the setDefault/reset method? If dimension and field count are set to zero, the polynomial will be no longer valid. I don't see why setDefault/reset should be called instead of calling clear. If, for performance reasons, you don't want to release the memory  hold by the polynomial, you can add a release argument to the clear method and pass that argument when you clear the polynomial. I would delete the method setDefault/reset.